packages/jest-core/src/__tests__/watch-file-changes.test.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import path from 'path';
+import fs from 'fs';
+import os from 'os';
+import {JestHook} from 'jest-watcher';
+import Runtime from 'jest-runtime';
+import {normalize} from 'jest-config';
+import HasteMap from 'jest-haste-map';
+import rimraf from 'rimraf';
+import {AggregatedResult} from '@jest/test-result';
+
+describe('Watch mode flows with changed files', () => {
+  jest.resetModules();
+
+  let watch: any;
+  let pipe: NodeJS.ReadStream;
+  let stdin: MockStdin;
+  const testDirectory = path.resolve(os.tmpdir(), 'jest-tmp');
+  const fileTargetPath = path.resolve(testDirectory, 'lost-file.js');
+  const fileTargetPath2 = path.resolve(
+    testDirectory,
+    'watch-test-fake.test.js',
+  );
+  const cacheDirectory = path.resolve(os.tmpdir(), `tmp${Math.random()}`);
+  let hasteMapInstance: HasteMap;
+
+  beforeEach(() => {
+    watch = require('../watch').default;
+    pipe = {write: jest.fn()} as any;
+    stdin = new MockStdin();
+    rimraf.sync(cacheDirectory);
+    rimraf.sync(testDirectory);
+    fs.mkdirSync(testDirectory);
+    fs.mkdirSync(cacheDirectory);
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+    if (hasteMapInstance) {
+      hasteMapInstance.end();
+    }
+    rimraf.sync(cacheDirectory);
+    rimraf.sync(testDirectory);
+  });
+
+  it('should correct require new files without legacy cache', async () => {
+    fs.writeFileSync(
+      fileTargetPath2,
+      `
+        require('./lost-file.js');
+        describe('Fake test', () => {
+            it('Hey', () => {  
+            
+            });
+        });
+      `,
+      {
+        encoding: 'utf-8',
+      },
+    );
+
+    const config = normalize(
```

Comment:
any reason not to use `createGlobalConfig` and `createProjectConfig` from `TestUtils.ts` in root here? Instead of `normalize`

Reply:
`makeGlobalConfig` from `TestUtils.ts` doesn't use normalizer inside. Without it this test will not work correct.